uaclient/system.py

```diff
@@ -59,96 +41,51 @@
     "KernelInfo",
     [
         ("uname_release", str),
-        ("proc_version_signature_full", str),
-        ("proc_version_signature_version", str),
-        ("version", str),
-        ("major", int),
-        ("minor", int),
-        ("patch", int),
-        ("abi", str),
-        ("subrev", str),
-        ("hwerev", str),
-        ("tag", str),
-        ("flavor", str),
+        ("proc_version_signature_version", Optional[str]),
+        ("major", Optional[int]),
+        ("minor", Optional[int]),
+        ("patch", Optional[int]),
+        ("abi", Optional[str]),
+        ("flavor", Optional[str]),
     ],
 )
 
 
 @lru_cache(maxsize=None)
 def get_kernel_info() -> KernelInfo:
-    uname_release = os.uname().release
-
-    proc_version_signature_full = ""
-    proc_version_signature_version = ""
+    proc_version_signature_version = None
     try:
         proc_version_signature_full = load_file("/proc/version_signature")
         proc_version_signature_version = proc_version_signature_full.split()[1]
     except Exception:
+        logging.warning("failed to process /proc/version_signature.")
+
+    uname_release = os.uname().release.strip()
+    match = re.match(RE_KERNEL_UNAME, uname_release)
```

Comment:
And just a nitpick,  that was probably already here in the code, but let's just rename that variable, since `match` seems to be a reserved name in python